File: tdd-money/money.py


class Dollar(object):
    def __init__(self, amount):
        self.amount = amount

    def times(self, multiplier):
        return Dollar(self.amount * multiplier)

    def __eq__(self, other):
        return self.amount == other.amount

    def equals(self, other):
        return self.amount == other.amount

    def kiss(self):
        return 0

File: tdd-java/src/money/Money.java
package money;

abstract class Money {
	protected int amount = 0;
	protected String currency = "";
	
	abstract Money times(int multiplier);
	
	Money(int amount, String currency) {
		this.amount = amount;
		this.currency = currency;
	}
	
	static Money dollar(int amount){
		return new Dollar(amount, "USD");
	}
	
	static Money franc(int amount) {
		return new Franc(amount, "CHF");
	}
	
	public String currency() {
		return currency;
	}
	
	public boolean equals(Object object) {
		Money money = (Money) object;
		return amount == money.amount &&
				getClass().equals(money.getClass());
	}
	

}
